routing/router.go

```diff
@@ -1723,6 +1723,186 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *Route
 	}
 }
 
+// SendToRoute attempts to send a payment as described within the passed
+// LightningPayment through the provided routes. This function is blocking
+// and will return either: when the payment is successful, or all routes
+// have been attempted and resulted in a failed payment. If the payment
+// succeeds, then a non-nil Route will be returned which describes the
+// path the successful payment traversed within the network to reach the
+// destination. Additionally, the payment preimage will also be returned.
+func (r *ChannelRouter) SendToRoute(routes []*Route,
```

Comment:
This adds a lot of code duplication. Instead should `SendPayment` be modified to use `SendToRoute` internally.

Reply:
@halseth thanks for reviewing! I've considered your suggestion but decided to create a private method `sendPayment` which is used by `SendPayment` and `SendToRoute`.